lib/src/clojure_lsp/queries.clj

```diff
@@ -15,14 +15,23 @@
     (.equals ^clojure.lang.Symbol a b)
     (.equals ^String a b)))
 
-(defn find-first [pred coll]
+(defn ^:private find-first [pred coll]
   (reduce
     (fn [_ i]
       (when (pred i)
         (reduced i)))
     nil
     coll))
 
+(defn ^:private project-analysis [{:keys [analysis]}]
+  (medley/remove-vals
+    (fn [elems]
+      (:external? (first elems)))
+    analysis))
+
+(defn db-with-project-analysis [db]
+  (assoc db :analysis (project-analysis db)))
+
```

Comment:
I think this is the only change that concerns me a little bit as we are creating a new concept of a db but where the analysis are not "full", and relies only on calling it as a `project-db`, I don't think it's a big deal, but maybe avoiding using it outside this namespace may reduce this concern? kind of related with https://github.com/clojure-lsp/clojure-lsp/pull/1040/files?diff=unified&w=1#r894925391

Reply:
I see what you mean @ericdallo, but I also think it's very similar to what we were doing before—filtering the analysis and passing that around, relying only calling it `project-analysis`.

I understand what you're saying about moving the usages into `queries`, but that'd be tricky in several situations. For example, consider custom-linting. We want to do all of our custom linting based on project files, excluding external files. That algorithm needs to run 4 queries on the analysis: to find the used vars and keywords, and then to find unused vars and keywords. We could move each of those queries into the `queries` namespace, but then we'd be filtering to the project analysis 4 times instead of once. Filtering to project analysis is much faster than it used to be, but still, I'd rather do it as few times as possible.

It's a similar situation for `file-management/find-references`, one of the other places that uses `queries/db-with-project-analysis`. We run a couple of queries on the project-db, and it'd be nice to filter only once. The final usage is in `internal-api/rename!*`. That one could pretty easily be moved into `queries`. I'll work on that.

But in general, this is hard to get away from. There are other places where we used to filter the analysis to a single file or a few files, and pass that around calling it `local-analysis` or something like that. This PR is doing the same thing, but with a filtered version of the db.

Anyway, you have a point—this is sort of a code smell, where perhaps we're exposing more than we want to. I'll keep that in mind as I work on dep-graph queries, which will include changes to many of the same locations. If I can, as I'm working on that, I'll move things into `queries`.

Reply:
One other thought... I think this code is also a little surprising because it introduces a new way to filter to project analysis, something we've been doing with `filter-project-analysis-xf` so far. I plan to revisit that eventually, and hopefully I'll be able to get back to having just one way of doing things. But for awhile the project will be in a weird state, where we have older parts of `queries` mixed with newer parts. 

I need to keep both parts in the code at the same time so that we can have a flag to toggle dep-graph queries. But when the flag is removed, I'll plan to clean things up.

Reply:
> filtering the analysis and passing that around, relying only calling it project-analysis.

Good point

> Anyway, you have a point—this is sort of a code smell

Yeah, exactly, this is not a new problem, it's something we are already doing in multiple places as you mentioned, I think it's good to keep that in mind during those refactors, for now I guess it's ok